numpy/core/src/multiarray/buffer.c

```diff
@@ -504,9 +504,37 @@ _buffer_info_new(PyObject *obj)
             info->shape = (npy_intp *)((char *)info + sizeof(_buffer_info_t));
             assert((size_t)info->shape % sizeof(npy_intp) == 0);
             info->strides = info->shape + PyArray_NDIM(arr);
-            for (k = 0; k < PyArray_NDIM(arr); ++k) {
-                info->shape[k] = PyArray_DIMS(arr)[k];
-                info->strides[k] = PyArray_STRIDES(arr)[k];
+
+            /*
+             * When NPY_RELAXED_STRIDES_CHECKING is used, some buffer users
+             * may expect a contiguous buffer to have well formatted strides
+             * also when a dimension is 1, but we do not guarantee this
+             * internally. Thus, recalculate strides for contiguous arrays.
+             * (This is unnecessary, but has no effect in the case where
+             * NPY_RELAXED_STRIDES CHECKING is disabled.)
+             */
```

Comment:
Do we have any evidence these users exist?

Reply:
Also, should this code be conditional on NPY_RELAXED_STRIDES_CHECKING ?

Reply:
Happy to make it conditional. Without it, its basically a no-op currently.

As to evidence: There used to be issues with scipy, etc. when we switched NumPy over.  From the comment, I somewhat think Python _used_ to have issues around it (I doubt it still has with a brief look at the code).

It would be nice to just remove this dance completely.  But, I am unwilling to do this in 1.19, so I think we should probably put this in anyway, and then consider doing the additional step with a big release note.  The problem is, that relaxed strides was not a simple matter: We had to do multiple releases with it enabled during RC, but disabled during release to flush out (potential) issues.  We can do that again of course (set relaxed strides to debug mode during release), but it might be painful.